azurelinuxagent/common/cgroupapi.py

```diff
@@ -423,14 +426,45 @@ def start_extension_command(self, extension_name, command, shell, cwd, env, stdo
             env=env,
             preexec_fn=os.setsid)
 
-        logger.info("Started extension using scope '{0}'", scope_name)
+        # Wait a bit and check if we completed with error
+        time.sleep(1)
+        return_code = process.poll()
+
+        if return_code is not None and return_code != 0:
+            # Check journal file for this scope and see if it started successfully
+            _, journal = shellutil.run_get_output("journalctl --unit={0}.scope".format(scope_name))
+
+            if "-- No entries --" in journal:
+                logger.warn('Failed to run systemd-run for unit {0}.'.format(scope_name))
```

Comment:
we need to capture the error and send that info to Kusto -- exception, error message, whatever we have avaialble

Reply:
Agreed, since that data is critical for debugging, I'll go ahead and abandon the `journalctl` approach and use `stdout` for checks. This then resolves your other concerns about `journalctl`.

Reply:
just a note: you may also need to capture the stderr of systemd-run

Reply:
I edited my previous comment before I saw yours :)